plaid-stl/src/npm/mod.rs

```diff
@@ -0,0 +1,607 @@
+use serde::{Deserialize, Deserializer, Serialize};
+
+use crate::PlaidFunctionError;
+
+use chrono::{DateTime, Utc};
+
+// All the structs which are also used in the Plaid API
+
+/// Permission that is granted to a team over an npm package
+#[derive(Serialize, Deserialize, Clone)]
+pub enum NpmPackagePermission {
+    READ,
+    WRITE,
+}
+
+impl ToString for NpmPackagePermission {
+    fn to_string(&self) -> String {
+        match self {
+            Self::READ => "read".to_string(),
+            Self::WRITE => "write".to_string(),
+        }
+    }
+}
+
+impl NpmPackagePermission {
+    // We do not implement the TryFrom trait because of the error type. We do not want to carry the
+    // NpmError struct here in the STL. Also, we don't care about which error we get from here. If
+    // it fails, the caller will know what to do.
+    pub fn try_from(value: String) -> Result<Self, ()> {
+        match value.as_str() {
+            "read" => Ok(NpmPackagePermission::READ),
+            "write" => Ok(NpmPackagePermission::WRITE),
+            _ => Err(()),
+        }
+    }
+}
+
+/// Teams in the npm organization
+#[derive(Serialize, Deserialize)]
+pub enum NpmTeam {
+    Admins,
+    Developers,
+}
+
+impl ToString for NpmTeam {
+    fn to_string(&self) -> String {
+        match self {
+            Self::Admins => "Admins".to_string(),
```

Comment:
One is capital the other is not

Reply:
Yep :)
But these are the names that are configured on the website, so we need those to appear in the requests we make

Reply:
Removed when moving to using strings instead of enum for teams.